packages/kit/src/core/dev/index.js

```diff
@@ -231,15 +231,31 @@ function get_params(array) {
 	// src/routes/[x]/[y]/[z]/svelte, create a function
 	// that turns a RegExpExecArray into ({ x, y, z })
 
+	// input has already been decoded by decodeURI
+	// now handle the rest that decodeURIComponent would do
+	const d = /** @param {string} s */ (s) =>
```

Comment:
This looks the same like in `build` - maybe put it into a common file for both to use? `utils/encoding.js` or something like that.

Reply:
Yeah, it is the same, but I'm not sure it's that easy to share code between the two because the one in `build` is a string that gets written out to a file and isn't normal code. I think I could still do it, but would probably need to setup a new file to be output in the Rollup config as well. I'm just not sure it's worth it at the moment given all the other stuff there is to do though I do agree it'd be nicer